scripts/linking.py

```diff
@@ -105,77 +182,87 @@ def create_load_tfidf_ann_index(model_path: str, umls_concept_list: List) -> Tup
     num_threads = 60  # set based on the machine
     index_params = {'M': M, 'indexThreadQty': num_threads, 'efConstruction': efC, 'post' : 0}
 
+    print(f'No tfidf vectorizer on {tfidf_vectorizer_path} or ann index on {ann_index_path}')
+    uml_concept_ids = []
+    uml_concept_aliases = []
+    print('Collecting aliases ... ')
+    for i, concept in enumerate(umls_concept_list):
+        concept_id = concept['concept_id']
 
+        # Alias lists for concepts are not unique, so calling set here means
+        # we don't duplicate items in the index. In practice this reduces the size
+        # of the index by 15%.
+        concept_aliases = list(set(concept['aliases'])) + [concept['canonical_name']]
```

Comment:
👍

Reply:
Is the `395MB` size of the tfidf matrix with or without this change ?

Reply:
This is with this change, but most of the saving comes from saving in half precision and using the sparse format.